webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -23,6 +30,69 @@ function CoreMenuOptions({handleOnSelect, channelId}: Props) {
         isDM,
     } = useTimePostBoxIndicator(channelId);
 
+    const recentlyUsedCustomDate = useSelector((state: GlobalState) => getPreference(state, scheduledPosts.SCHEDULED_POSTS, scheduledPosts.RECENTLY_USED_CUSTOM_TIME));
```

Comment:
We are adding a lot of logic just for one option. What do you think about moving all this to a different component?

Reply:
I think for now it can still live here, component is still small. In case more changes are needed on this part of the code, will keep in mind this suggestion.